go/teams/multiple_use_invite_test.go

```diff
@@ -192,3 +212,197 @@ func TestSeitanHandleExceededInvite(t *testing.T) {
+	t.Logf("Created team %s", teamName)
+
+	// Add team invite link which expires in an hour.
+	expTime := keybase1.ToUnixTime(clock.Now().Add(10 * time.Minute))
+	invLink, err := CreateInvitelink(tc.MetaContext(), teamName.String(), keybase1.TeamRole_READER, keybase1.TeamMaxUsesInfinite,
+		&expTime)
+	require.NoError(t, err)
+
+	API := libkb.NewAPIArgRecorder(tc.G.API)
+	tc.G.API = API
+
+	// Accept the link as user2.
+	kbtest.LogoutAndLoginAs(tc, user2)
+
+	uv := user2.GetUserVersion()
+	unixNow := clock.Now().Unix()
+	accepted, err := generateAcceptanceSeitanInviteLink(invLink.Ikey, uv, unixNow)
+	require.NoError(t, err)
+
+	err = postSeitanInviteLink(tc.MetaContext(), accepted)
+	require.NoError(t, err)
+
+	// This simulates a seitan msg for the admin as it would have came from
+	// team_rekeyd.
+	origMsg := keybase1.TeamSeitanMsg{
+		TeamID: teamID,
+		Seitans: []keybase1.TeamSeitanRequest{
+			{
+				InviteID:    keybase1.TeamInviteID(accepted.inviteID),
+				Uid:         uv.Uid,
+				EldestSeqno: uv.EldestSeqno,
+				Akey:        keybase1.SeitanAKey(accepted.encoded),
+				Role:        keybase1.TeamRole_READER,
+				UnixCTime:   unixNow,
+			},
+		},
+	}
+
+	adminCallsHandleTeamSeitanAndReturnsRejectCalls := func(msg keybase1.TeamSeitanMsg) []libkb.APIRecord {
+		kbtest.LogoutAndLoginAs(tc, admin)
+		err = HandleTeamSeitan(context.TODO(), tc.G, msg)
+		require.NoError(t, err)
+		return API.GetFilteredRecordsAndReset(func(rec *libkb.APIRecord) bool {
+			return rec.Arg.Endpoint == "team/reject_invite_acceptance"
+		})
+	}
+
+	assertRejectInviteArgs := func(record libkb.APIRecord, inviteID SCTeamInviteID, uid keybase1.UID, seqno keybase1.Seqno) {
+		require.Equal(t, string(inviteID), record.Arg.Args["invite_id"].String())
+		require.Equal(t, string(uid), record.Arg.Args["uid"].String())
+		require.Equal(t, fmt.Sprintf("%v", seqno), record.Arg.Args["eldest_seqno"].String())
+	}
+
+	// change the eldest seqno and ensure the invite is rejected
+	fakeMsg := origMsg.DeepCopy()
+	fakeMsg.Seitans[0].EldestSeqno = 5
+	records := adminCallsHandleTeamSeitanAndReturnsRejectCalls(fakeMsg)
+	require.Len(t, records, 1, "one invite acceptance should be rejected")
+	record := records[0]
+	assertRejectInviteArgs(record, accepted.inviteID, uv.Uid, keybase1.Seqno(5))
+	// since this modified invite acceptance (has different eldestSeqno) was
+	// never sent to the server, rejection should fail
+	require.Contains(t, record.Err.Error(), "acceptance not found")
+
+	// now change the akey to something that cannot be b64 decoded
+	fakeMsg = origMsg.DeepCopy()
+	fakeMsg.Seitans[0].Akey = keybase1.SeitanAKey("*") + fakeMsg.Seitans[0].Akey[1:]
+	records = adminCallsHandleTeamSeitanAndReturnsRejectCalls(fakeMsg)
+	require.Len(t, records, 1, "one invite acceptance should be rejected")
+	record = records[0]
+	assertRejectInviteArgs(record, accepted.inviteID, uv.Uid, uv.EldestSeqno)
+	// We do not test the rejection error here, as team_rekeyd might or might
```

Comment:
maybe change the underlying api to a mock that drops the requests, so you don't have to do this re-post every time